Source/Core/UICommon/UICommon.cpp

```diff
@@ -309,24 +309,25 @@ void SetUserDirectory(std::string custom_path)
     // We are on a non-Apple and non-Android POSIX system, there are 4 cases:
     // 1. GetExeDirectory()/portable.txt exists
     //    -> Use GetExeDirectory()/User
-    // 2. $DOLPHIN_EMU_USERPATH is set
+    // 2. $DOLPHIN_EMU_USERPATH is set, and we're not in flatpak
     //    -> Use $DOLPHIN_EMU_USERPATH
-    // 3. ~/.dolphin-emu directory exists
+    // 3. ~/.dolphin-emu directory exists, and we're not in flatpak
     //    -> Use ~/.dolphin-emu
     // 4. Default
     //    -> Use XDG basedir, see
     //    http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html
     user_path = home_path + "." DOLPHIN_DATA_DIR DIR_SEP;
     std::string exe_path = File::GetExeDirectory();
+    bool flatpak = File::Exists("/.flatpak-info");
     if (File::Exists(exe_path + DIR_SEP "portable.txt"))
     {
       user_path = exe_path + DIR_SEP "User" DIR_SEP;
     }
-    else if (env_path)
+    else if (env_path && !flatpak)
```

Comment:
Shouldn't `DOLPHIN_EMU_USERPATH` be available as an override?
Even when `portable.txt` exists?

Reply:
My concern was that we'd get bug reports from users who had unknowingly set the env variable (by installing from aur/copr/ppa/etc) to an unwritable path.

Of course, we could also deal with this by unsetting the variable in the manifest, which would produce the same result. Do you believe it'd be better?

Reply:
I think it would be better of in the manifest if that works.
If the user wants to use the variable they'll be fishing around in the flatpak overrides to set filesystem permissions anyway.

Reply:
It's a very a good point, sorry to take so long, I've fixed it and I'll rebase and push the branch now